File: front-end/src/app/services/urls.ts
import http from "./service";
import { User } from "../interface/interface";

class URLS {
  getCsrfToken() {
    return http.get("/csrf");
  }

  Register(user: User) {
    return http.post("/register", user);
  }

  login(user: User) {
    return http.post("/login", user);
  }
  doLogout() {
    return http.post("/logout");
  }

  getOneUser(id: any) {
    return http.get(`/getOneUser/${id}`);
  }
  
  getUser() {
    return http.get("getUser");
  }

  getAll() {
    return http.get("movies");
  }
  getById(id: any) {
    return http.get(`get/${id}`);
  }

  addReview(id: any, review: any) {
    return http.post(`/post/${id}`, review);
  }

  getReviews() {
    // noStore();
    return http.get(`/reviews`);
  }
  watch_list(id: any){
    return http.post(`${id}/watch_list`);
  }
  userWatch(){
    return http.get(`watch_list`)
  }
  rating_film(id: any){
    return http.get(`/rating_film/${id}`);
  }

  created_rate(id: any, rating: any){
    return http.post(`/created_rate/${id}`, rating);
  }
}
export default new URLS();

File: BackEnd/cinema/urls.py
from django.urls import path

from . import views

urlpatterns = [
    path("movies", views.index, name="index"),
    path("login", views.login_view, name="login"),
    path("logout", views.logout_view, name="logout"),
    path("register", views.register, name="register"),
    path("getOneUser/<str:id>",views.oneUser, name="oneUser"),
    path("getUser",views.ALLUser, name="AllUser"),
    path("get/<str:id>",views.oneMovie, name="oneMovie"),
    path("reviews",views.list_reviews,name="listOfReviews"),
    path("post/<str:film_id>",views.created_review, name="createdReview"),
    path("csrf", views.get_csrf, name="csrftoken"),
    path("<str:watch_id>/watch_list", views.watch_list, name= "watchList"),
    path("watch_list", views.favored, name = "watch_list"),
    path("rating_film/<str:film_id>", views.rating_film, name = "ratingFilm"),
    path("created_rate/<str:film_id>", views.created_rate, name = "createdRate"),
]